overlay/utils.sh

```diff
@@ -1,206 +1,283 @@
+# shellcheck shell=bash
+
 extractFileExt() {
-    local name=`basename $1`
-    echo ${name##*.}
+  local name=$(basename "$1")
+  echo "${name##*.}"
 }
+
 extractHash() {
-    local name=`basename $1`
-    echo ${name%%-*}
+  local name=$(basename "$1")
+  echo "${name%%-*}"
```

Comment:
Super minor nitpick, but I wonder if leaving basename is better? Definitely bikeshedding but I had to go off and read [this](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02) and [this](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02) to make sure that that's correct. That or leave a comment for people unfamilar with bash pattern expansion syntax?

Reply:
We can't use `basename` alone because that would produce `crate-foo-X.Y.Z-################` and we need only `################`. Sure, we could add a comment here, but the purpose of this Bash pattern is ultimately made redundant by the function name itself, `extractHash`.